plugins/outputs/arc/arc_test.go

```diff
@@ -0,0 +1,446 @@
+			name: "valid config",
+			arc: &Arc{
+				URL:              "http://localhost:8000/api/v1/write/msgpack",
+				HTTPClientConfig: common_http.HTTPClientConfig{Timeout: config.Duration(5 * time.Second)},
+			},
+			expectError: false,
+		},
+		{
+			name: "default values",
+			arc: &Arc{
+				URL: "",
+			},
+			expectError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.arc.Init()
+			if tt.expectError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestConnect(t *testing.T) {
+	plugin := &Arc{
+		URL:              "http://localhost:8000/api/v1/write/msgpack",
+		HTTPClientConfig: common_http.HTTPClientConfig{Timeout: config.Duration(5 * time.Second)},
+		Log:              testutil.Logger{},
+	}
+
+	require.NoError(t, plugin.Init())
+	require.NoError(t, plugin.Connect())
+	require.NotNil(t, plugin.client)
+}
+
+func TestWrite(t *testing.T) {
+	tests := []struct {
+		name            string
+		metrics         []telegraf.Metric
+		expectedRecords int
+		contentEncoding string
+	}{
+		{
+			name: "single metric",
+			metrics: []telegraf.Metric{
+				metric.New(
+					"cpu",
+					map[string]string{
+						"host": "server01",
+						"cpu":  "cpu0",
+					},
+					map[string]interface{}{
+						"usage_idle":   float64(95.5),
+						"usage_system": float64(2.5),
+						"usage_user":   float64(2.0),
+					},
+					time.Unix(1633024800, 0),
+				),
+			},
+			expectedRecords: 1,
+			contentEncoding: "identity",
+		},
+		{
+			name: "multiple metrics with gzip",
+			metrics: []telegraf.Metric{
+				metric.New(
+					"cpu",
+					map[string]string{"host": "server01"},
+					map[string]interface{}{"usage_idle": float64(95.5)},
+					time.Unix(1633024800, 0),
+				),
+				metric.New(
+					"cpu",
+					map[string]string{"host": "server02"},
+					map[string]interface{}{"usage_idle": float64(85.0)},
+					time.Unix(1633024801, 0),
+				),
+			},
+			expectedRecords: 2,
+			contentEncoding: "gzip",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var receivedMethod string
+			var receivedContentType string
+			var receivedUserAgent string
+			var receivedContentEncoding string
+			var receivedBody []byte
+
+			// Create test server
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				receivedMethod = r.Method
+				receivedContentType = r.Header.Get("Content-Type")
+				receivedUserAgent = r.Header.Get("User-Agent")
+				receivedContentEncoding = r.Header.Get("Content-Encoding")
+
+				// Read body
+				var body io.Reader = r.Body
+				if receivedContentEncoding == "gzip" {
+					gz, err := gzip.NewReader(r.Body)
+					if err != nil {
+						t.Fail()
+						return
+					}
+					defer gz.Close()
+					body = gz
+				}
+
+				data, err := io.ReadAll(body)
+				if err != nil {
+					t.Fail()
+					return
+				}
+				receivedBody = data
+
+				w.WriteHeader(http.StatusNoContent)
```

Comment:
I think this might be racy with the tests as the http-server runs in its own go-routine IIRC. How about


```suggestion
			var receivedMethod string
			var receivedContentType string
			var receivedUserAgent string
			var receivedContentEncoding string
			var receivedBody []byte
			var receivedMu sync.Mutex
			var done atomic.Bool

			// Create test server
			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
				receivedMu.Lock()
				defer receivedMu.Unlock()
				
				receivedMethod = r.Method
				receivedContentType = r.Header.Get("Content-Type")
				receivedUserAgent = r.Header.Get("User-Agent")
				receivedContentEncoding = r.Header.Get("Content-Encoding")

				// Read body
				var body io.Reader = r.Body
				if receivedContentEncoding == "gzip" {
					gz, err := gzip.NewReader(r.Body)
					if err != nil {
						t.Fail()
						return
					}
					defer gz.Close()
					body = gz
				}

				data, err := io.ReadAll(body)
				if err != nil {
					t.Fail()
					return
				}
				receivedBody = data
				done.Store(true)

				w.WriteHeader(http.StatusNoContent)
```